snippets/lean.json

```diff
@@ -0,0 +1,48 @@
+{
+    "lemma": {
+        "prefix": "lemma",
+        "body": [
+            "lemma ${1:name (args)} : ${2:sorry} := ${3:sorry}"
+        ],
+        "description": "Insert a one-line lemma"
+    },
+    "theorem": {
+        "prefix": "theorem",
+        "body": [
+            "theorem ${1:name (args)} : ${2:sorry} := ${3:sorry}"
+        ],
+        "description": "Insert a one-line lemma"
+    },
+    "example": {
+        "prefix": "example",
+        "body": [
+            "example : ${2:sorry} := ${2:sorry}"
+        ],
+        "description": "Insert a one-line example"
+    },
+	"namespace": {
+        "prefix": "namespace",
+        "body": [
+            "namespace ${1:name}",
+            "$0",
+            "end ${1}"
+        ],
+        "description": "Insert a namespace"
+    },
+	"structure": {
+        "prefix": "structure",
+        "body": [
+            "structure ${1:name} ${2:(args : sorry)}",
+            ":=",
+            "\t${3:(field : sorry)}",
+            "",
+            "namespace ${1:}",
```

Comment:
The lines below are a little too much. `structure` should not make too much assumption on the organization after the structure. If this snippet is called `theory` for boilerplates of developing a theory, maybe it would be OK but there would be much more.